torch/csrc/autograd/engine.cpp

```diff
@@ -808,6 +765,45 @@ variable_list Engine::execute_with_graph_task(
   return graph_task->captured_vars_;
 }
 
+std::shared_ptr<FutureVariableList> Engine::execute_with_graph_task(
+    std::shared_ptr<GraphTask> graph_task,
+    std::shared_ptr<Node> graph_root) {
+  std::call_once(start_threads_flag_, &Engine::start_threads, this);
+  // Lock mutex for GraphTask.
+  std::unique_lock<std::mutex> lock(graph_task->mutex_);
+
+  ready_queue(at::kCPU).push(
+      NodeTask(graph_task, std::move(graph_root), InputBuffer(0)));
+
+  // Not a worker
+  if (worker_device == NO_DEVICE) {
+    return graph_task->future_;
```

Comment:
This would be skipping the post-processing in this case?

Reply:
It looks like this is intentional, right? The post-processing is supposed to be done by the entity fulfilling the future (i.e. mark_graph_task_completed), rather than after the future is fulfilled.

Semi-related, I am mildly questioning why we're holding the graph_task->mutex_ lock across post-processing (if nothing else, it might be nice to drop the lock right before calling graph_task->future_->markCompleted(vval)), but this doesn't necessarily look like an actual lock scope change vs. previousl.

Reply:
@albanD As @jjlilley mentioned this is done as part of `mark_graph_task_completed`. Regarding the locking, I just kept the same behavior as we already had. Before this PR all the post processing was run under the `graph_task->mutex_`.